app/forms/provider_interface/offer_wizard.rb

```diff
@@ -84,30 +84,33 @@ def next_step(step = current_step)
 
     delegate :previous_step, to: :wizard_path_history
 
-    def condition_models
-      @_condition_models ||= further_conditions.map.with_index do |further_condition, index|
-        OfferConditionField.new(id: index, text: further_condition)
+    def further_condition_models
+      @_further_condition_models ||= further_condition_attrs.map do |index, hash|
+        OfferConditionField.new(id: index.to_i, text: hash['text'], condition_id: hash['condition_id'])
       end
     end
 
     def has_max_number_of_further_conditions?
-      further_conditions.length >= MAX_FURTHER_CONDITIONS
+      further_condition_attrs.length >= MAX_FURTHER_CONDITIONS
     end
 
     def add_empty_condition
       return if has_max_number_of_further_conditions?
 
-      further_conditions << ''
+      further_condition_attrs.merge!(further_condition_attrs.length.to_s => { 'text' => '' })
       save_state!
     end
 
     def remove_condition(condition_id)
-      further_conditions.delete_at(condition_id)
+      further_condition_attrs.delete(condition_id)
+      further_condition_attrs.transform_keys!.with_index { |_key, index| index.to_s }
```

Comment:
Similar to below: https://github.com/DFE-Digital/apply-for-teacher-training/pull/4936/files#r650058644